code/datums/status_effects/buffs.dm

```diff
@@ -348,9 +348,19 @@
 
 /datum/status_effect/antimagic
 	id = "antimagic"
+	status_type = STATUS_EFFECT_REFRESH
 	duration = 10 SECONDS
 	examine_text = "<span class='notice'>They seem to be covered in a dull, grey aura.</span>"
 
+/datum/status_effect/antimagic/on_creation(mob/living/new_owner, duration = 10 SECONDS)
+	src.duration = duration
+	return ..()
+
+/datum/status_effect/antimagic/refresh(effect, duration = 10 SECONDS)
+	if(duration == -1)
```

Comment:
why exactly -1?

Reply:
-1 is how status effects set themselves to never expire